tests/unit/geo.tcl

```diff
@@ -42,6 +43,81 @@ proc compare_lists {List1 List2} {
    return $DiffList
 }
 
+# If a point in circular
+proc pointInCircular {radius_km lon lat search_lon search_lat} {
+    set radius_m [expr {$radius_km*1000}]
+    set distance [geo_distance $lon $lat $search_lon $search_lat]
+    if {$distance < $radius_m} {
+        return true
+    }
+    return false
+}
+
+# Judge whether a point is in the trapezoid.
+proc pointInBox {width_km height_km lon lat search_lon search_lat error} {
+    set width_m [expr {$width_km*1000*$error/2}]
+    set height_m [expr {$height_km*1000*$error/2}]
+    set lat_delta [geo_raddeg [expr {$height_m/6372797.560856}]]
+    set long_delta_top [geo_raddeg [expr {$width_m/6372797.560856/cos([geo_degrad [expr {$search_lat+$lat_delta}]])}]]
+    set long_delta_middle [geo_raddeg [expr {$width_m/6372797.560856/cos([geo_degrad $search_lat])}]]
+    set long_delta_bottom [geo_raddeg [expr {$width_m/6372797.560856/cos([geo_degrad [expr {$search_lat-$lat_delta}]])}]]
+
+    set bounds(0) [expr {$search_lon-$long_delta_top}]
+    set bounds(1) [expr {$search_lat+$lat_delta}]
+    set bounds(2) [expr {$search_lon+$long_delta_top}]
+    set bounds(3) [expr {$search_lat+$lat_delta}]
+    set bounds(4) [expr {$search_lon+$long_delta_middle}]
+    set bounds(5) $search_lat
+    set bounds(6) [expr {$search_lon+$long_delta_bottom}]
+    set bounds(7) [expr {$search_lat-$lat_delta}]
+    set bounds(8) [expr {$search_lon-$long_delta_bottom}]
+    set bounds(9) [expr {$search_lat-$lat_delta}]
+    set bounds(10) [expr {$search_lon-$long_delta_middle}]
+    set bounds(11) $search_lat
```

Comment:
@yangbodong22011 what do you think of this?
if that algorithm indeed works, maybe instead of using it in the tests we should use it in redis itself?
i initially suggested it just so that we don't use the same algo in the tests as the one in redis (weak test), but if we prove that it's a valid algo, it might be more accurate than the polygon one.

Reply:
@oranagra Thank you very much for your algorithm, i finished it at: https://github.com/redis/redis/pull/8445